src/main/java/se/lth/cs/connect/routes/Account.java

```diff
@@ -141,6 +147,57 @@ protected void setup(PippoSettings conf) {
             }
         });
 
+        // GET api.serp.se/v1/account/friends HTTP/1.1
+        GET("/friends", (rc)->{
+        	final JcNode coll = new JcNode("coll");
+
+        	//find all collections that a user have
+        	String email = rc.getParameter("email").toString();
+        	List<GrNode> res = Database.query(rc.getLocal("db"), new IClause[]{
+        	            MATCH.node().label("user").property("email").value(email)
+        	                .relation().type("MEMBER_OF").node(coll).label("collection"),
+        	            RETURN.DISTINCT().value(coll)
+        	        }).resultOf(coll);
+
+        	Set<GrNode> total = new HashSet<GrNode>();
+        	
+        	JcNode u = new JcNode("u");
+        	boolean found;
+        	//loop through all those collections and add all members inside the collections
+        	for(int i=0; i<res.size(); i++){
+            	List<GrNode> res2 = Database.query(rc.getLocal("db"), new IClause[]{
+       	        	 MATCH.node(u).label("user")
+       	             .relation().type("MEMBER_OF")
+       	             .node(coll).label("collection"),
+       		         WHERE.valueOf(coll.id()).EQUALS(res.get(0).getId()),
+       		         RETURN.DISTINCT().value(u)
+       	        }).resultOf(u);
+            	
+            	found = false;
+             	for(int j=0; j<res2.size(); j++){
+                	Iterator<GrNode> it = total.iterator();
+                	while(it.hasNext()){
+                		if(it.next().getId()==res2.get(j).getId()){
+                			found=true;
+                			break;
+                		}
+                	}
+                	if(!found){
+                		total.add(res2.get(j));
+                	}
+             	}
+        	}
+
+        	Iterator<GrNode> it = total.iterator();
+        	String[] ret = new String[total.size()];
+        	int i=0;
+        	while(it.hasNext()){
+        		ret[i]=it.next().getProperty("email").toString().split("= ")[1];
```

Comment:
because apparently proptery.toString() replies property = "email" and I only want the email. So i split it out.